cmd/check_test.go

```diff
@@ -236,3 +236,134 @@ p {
 		t.Fatalf("unexpected error from eval with inlined schema, got: %v", err)
 	}
 }
+
+func TestCheckRegoV1(t *testing.T) {
+	cases := []struct {
+		note    string
+		policy  string
+		expErrs []string
+	}{
+		{
+			note: "rego.v1 imported, v1 compliant",
+			policy: `package test
+import rego.v1
+p contains x if {
+	x := [1,2,3]
+}`,
+		},
+		{
+			note: "rego.v1 imported, NOT v1 compliant (parser)",
+			policy: `package test
+import rego.v1
+p contains x {
+	x := [1,2,3]
+}
+
+q.r`,
+			expErrs: []string{
+				"test.rego:3: rego_parse_error: `if` keyword is required before rule body",
+				"test.rego:7: rego_parse_error: `contains` keyword is required for partial set rules",
+			},
+		},
+		{
+			note: "rego.v1 imported, NOT v1 compliant (compiler)",
+			policy: `package test
+import rego.v1
+
+import data.foo
+import data.bar as foo
+`,
+			expErrs: []string{
+				"test.rego:5: rego_compile_error: import must not shadow import data.foo",
+			},
+		},
+		{
+			note: "keywords imported, v1 compliant",
+			policy: `package test
+import future.keywords.if
+import future.keywords.contains
+p contains x if {
+	x := [1,2,3]
+}`,
+		},
+		{
+			note: "keywords imported, NOT v1 compliant",
+			policy: `package test
+import future.keywords.contains
+p contains x {
+	x := [1,2,3]
+}
+
+q.r`,
+			expErrs: []string{
+				"test.rego:3: rego_parse_error: `if` keyword is required before rule body",
+				"test.rego:7: rego_parse_error: `contains` keyword is required for partial set rules",
+			},
+		},
+		{
+			note: "keywords imported, NOT v1 compliant (compiler)",
+			policy: `package test
+import future.keywords.if
+
+input := 1 if {
+	1 == 2
+}`,
+			expErrs: []string{
+				"test.rego:4: rego_compile_error: rules must not shadow input (use a different rule name)",
+			},
+		},
+		{
+			note: "no imports, v1 compliant",
+			policy: `package test
+p := 1
+`,
+		},
+		{
+			note: "no imports, NOT v1 compliant but v0 compliant (compiler)",
+			policy: `package test
+p.x`,
+			expErrs: []string{
+				"test.rego:2: rego_parse_error: `contains` keyword is required for partial set rules",
+			},
+		},
+		{
+			note: "no imports, v1 compliant but NOT v0 compliant",
+			policy: `package test
+p contains x if {
+	x := [1,2,3]
+}`,
+			expErrs: []string{
+				"test.rego:2: rego_parse_error: var cannot be used for rule name", // This error actually appears three times: once for 'p'; once for 'contains'; and once for 'x'. All are interpreted as [invalid] rule declarations with no value and body.
+				"test.rego:2: rego_parse_error: `if` keyword is required before rule body",
```

Comment:
This error seems weird 🤔

Reply:
Also if this policy is v1.0 compliant and `rego-v1` enabled, why do we generate an error here? If the flag wasn't set then this would be an error. If the purpose of the flag is to check for Rego v1 compatibility then this should not be an error, correct? Or if this is not the expected behavior, then let's point this out in the command description.

Reply:
Hm, yeah, the `if` error might be a bit confusing here .. I'll see if there is a way around it.

Reply:
The `--rego-v1` flag actually doesn't check for _only_ 1.0 compatibility. We still need the policy to also be compatible with the _ current_ 0.x version of OPA, which implies either `rego.v1` or `future.keywords` imports (which won't be requirements in 1.0). I'll update the description for the flag.